### PR TITLE
feat: DNS zone commandlet and some error case improvements

### PR DESCRIPTION
**Pull request by rjaini**
DNS zone commandlet and some error case improvements

### CODE REVIEW

File: registry/diagnosis/collectlogs.sh [hunk 1]
```diff
@@ -37,13 +37,25 @@ done
 
 mkdir -p ${LOGDIRECTORY}/containers
 echo "[$(date +%Y%m%d%H%M%S)][INFO] Collecting registry container logs."
+
+i=0
 for cid in $(sudo docker ps -a -q --no-trunc)
 do
+    echo "[$(date +%Y%m%d%H%M%S)][INFO] Getting cname of ${cid}."
```
Comment (jadarsie) [inline registry/diagnosis/collectlogs.sh]: `cname` is just a variable name, `name` would be better

File: registry/Scripts/pre-reqs.ps1 [hunk 2]
```diff
@@ -65,6 +68,173 @@ function New-ResourceGroup (
...
+    # Create storage account
+    
+    Write-Host "Check if DNS Zone ($Name) already exists"
+    $ErrorActionPreference = "SilentlyContinue";
+    $dnsZoneDetails = Get-AzureRmDnsZone -Name $Name `
+                                         -ResourceGroupName $ResourceGroupName `
+                                         -ErrorVariable dnsZoneExistError
+    $ErrorActionPreference = "Continue"; #Turning errors back on
+    if (-not $dnsZoneDetails) {
+        Write-Host "DNS zone does not exists. Creating a new DNS zone ($Name) under resource group ($ResourceGroupName)."
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: no `s` here, `does not exist`

File: registry/Scripts/pre-reqs.ps1 [hunk 3]
```diff
@@ -65,6 +68,173 @@ function New-ResourceGroup (
...
+    $ErrorActionPreference = "Continue"; #Turning errors back on
+    if (-not $dnsZoneDetails) {
+        Write-Host "DNS zone does not exists. Creating a new DNS zone ($Name) under resource group ($ResourceGroupName)." 
+        New-AzureRmDnsZone -Name $Name -ResourceGroupName $ResourceGroupName | Out-Null
+        #$ErrorActionPreference = "SilentlyContinue";
+        $newDNSZoneDetails = Get-AzureRmDnsZone -Name $Name -ResourceGroupName $ResourceGroupName `
+                                                       -ErrorVariable newDNSZoneCreateError
+        #$ErrorActionPreference = "Continue"; #Turning errors back on
+        if  (-not $newDNSZoneDetails) {
+            throw "Creation a new DNS zone ($Name) under resource group ($ResourceGroupName) failed. Check if specified DNS zone already exists under a subscription."
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: `creation of` or `creating a`
capital Z, `Zone`
`under a subscription` is not clear

File: registry/Scripts/pre-reqs.ps1 [hunk 4]
```diff
@@ -65,6 +68,173 @@ function New-ResourceGroup (
...
+    [string] $ResourceGroupName,
+    [Parameter(Mandatory = $true, HelpMessage = "DNS zone name.")]
+    [string] $ZoneName,
+    [Parameter(Mandatory = $true, HelpMessage = "DNS record name.")]
+    [string] $Name,
+    [Parameter(Mandatory = $true, HelpMessage = "Type of record.")]
+    [string] $RecordType,
+    [Parameter(Mandatory = $true, HelpMessage = "Record value.")]
+    [string] $DnsRecordConfigValue,
+    [Parameter(Mandatory = $false, HelpMessage = "Ttl value.")]
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: `TTL`

File: registry/diagnosis/collectlogs.sh (hunk 1 above)
Comment (rjaini) [reply registry/diagnosis/collectlogs.sh]: Sure. change it to container name.

File: registry/Scripts/pre-reqs.ps1 (hunk 2 above)
Comment (rjaini) [reply registry/Scripts/pre-reqs.ps1]: sure.

File: registry/Scripts/pre-reqs.ps1 (hunk 3 above)
Comment (rjaini) [reply registry/Scripts/pre-reqs.ps1]: updated to Check if specified DNS zone with same name already exists under the selected subscription

File: registry/Scripts/pre-reqs.ps1 (hunk 4 above)
Comment (rjaini) [reply registry/Scripts/pre-reqs.ps1]: I used to align the way commandlet is taking input. But will update.

**Pull request merged**